Framework/Algorithms/test/Q1DWeightedTest.h

```diff
@@ -300,6 +303,101 @@ class Q1DWeightedTest : public CxxTest::TestSuite {
+        radial_average.setPropertyValue("WedgeWorkspace", outputWS + "_wedges"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("AsymmetricWedges", "1"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("ShapeTable", "MaskShapes"))
+
+    TS_ASSERT_THROWS_NOTHING(radial_average.execute())
+
+    WorkspaceGroup_sptr result;
+    TS_ASSERT_THROWS_NOTHING(
+        result = std::dynamic_pointer_cast<WorkspaceGroup>(
+            AnalysisDataService::Instance().retrieve(outputWS + "_wedges")))
+    TS_ASSERT_EQUALS(result->getNumberOfEntries(), 2)
+  }
+
+  void testShapeTableResults() {
+    // test the results computed by the table shape method against those from
+    // the usual method
+
+    createShapeTable(true);
+
+    std::string outputWS = "q1d_shapes";
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("InputWorkspace", m_inputWS))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("OutputWorkspace", outputWS))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("WedgeWorkspace", outputWS + "_wedges"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("OutputBinning", "0.001,0.001,0.08"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("AsymmetricWedges", "0"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("ShapeTable", "MaskShapes"))
+    TS_ASSERT_THROWS_NOTHING(radial_average.execute())
+
+    std::string refWS = "q1d_wedges";
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("InputWorkspace", m_inputWS))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("OutputWorkspace", refWS))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("OutputBinning", "0.001,0.001,0.08"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("NumberOfWedges", "2"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("WedgeAngle", "90"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("WedgeOffset", "0"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("AsymmetricWedges", "0"))
+    TS_ASSERT_THROWS_NOTHING(
+        radial_average.setPropertyValue("WedgeWorkspace", refWS + "_wedges"))
+
+    TS_ASSERT_THROWS_NOTHING(radial_average.execute())
+
+    WorkspaceGroup_sptr result;
+    TS_ASSERT_THROWS_NOTHING(
+        result = std::dynamic_pointer_cast<WorkspaceGroup>(
+            AnalysisDataService::Instance().retrieve(outputWS + "_wedges")))
+
+    WorkspaceGroup_sptr ref = std::dynamic_pointer_cast<WorkspaceGroup>(
+        AnalysisDataService::Instance().retrieve(refWS + "_wedges"));
+
+    auto wedge1 =
+        std::dynamic_pointer_cast<MatrixWorkspace>(result->getItem(0));
+    auto wedge2 =
+        std::dynamic_pointer_cast<MatrixWorkspace>(result->getItem(1));
+    TS_ASSERT(wedge1)
+    TS_ASSERT(wedge2)
+
```

Comment:
Here you should use `CompareWorkspaces` to check everything (that make sense) rather than just the intensities.